engine/src/main/scala/cromwell/engine/workflow/lifecycle/execution/preparation/InputPointerToWdlValue.scala

```diff
@@ -2,31 +2,33 @@ package cromwell.engine.workflow.lifecycle.execution.preparation
 
 import cats.syntax.validated._
 import cromwell.core.ExecutionIndex.ExecutionIndex
-import cromwell.engine.workflow.lifecycle.execution.OutputStore
+import cromwell.engine.workflow.lifecycle.execution.ValueStore
 import lenthall.validation.ErrorOr.ErrorOr
 import shapeless.Poly1
 import wdl.values.WdlValue
 import wom.expression.{IoFunctionSet, WomExpression}
+import wom.graph.GraphNode
 import wom.graph.GraphNodePort.OutputPort
 
 object InputPointerToWdlValue extends Poly1 {
   // Function that can transform any of the coproduct types to an ErrorOr[WdlValue]
-  type ToWdlValueFn = (Map[String, WdlValue], IoFunctionSet, OutputStore, ExecutionIndex) => ErrorOr[WdlValue]
+  type ToWdlValueFn = (GraphNode, Map[String, WdlValue], IoFunctionSet, ValueStore, ExecutionIndex) => ErrorOr[WdlValue]
 
   implicit def fromWdlValue: Case.Aux[WdlValue, ToWdlValueFn] = at[WdlValue] {
-    wdlValue => (_: Map[String, WdlValue], _: IoFunctionSet, _: OutputStore, _: ExecutionIndex) =>
+    wdlValue => (_: GraphNode, _: Map[String, WdlValue], _: IoFunctionSet, _: ValueStore, _: ExecutionIndex) =>
       wdlValue.validNel: ErrorOr[WdlValue]
   }
 
   implicit def fromOutputPort: Case.Aux[OutputPort, ToWdlValueFn] = at[OutputPort] {
-    port => (_: Map[String, WdlValue], _: IoFunctionSet, outputStore: OutputStore, index : ExecutionIndex) =>
-      outputStore
-        .get(port, index).map(_.validNel)
+    port => (node: GraphNode, _: Map[String, WdlValue], _: IoFunctionSet, valueStore: ValueStore, index : ExecutionIndex) =>
+      valueStore.get(port, index)
+        .orElse(valueStore.get(port, None))  
```

Comment:
❓

Reply:
yeah this is what I was trying to explain at the tech talk last time.
When we're trying to get the value for another port from the value store, we don't always have enough information to know if we should get the one at the index we have or at index None. This is because we don't know if we're referring to something in the same scatter as we are, or outside of it. However it should never be at both at the same time so this tries one and then the other.
It's clunky though and I really don't like that, I just didn't have time to figure this out. It should definitely be cleaned up. I'll add a ticket.